modules/subnets-beta/main.tf

```diff
@@ -52,16 +52,14 @@ resource "google_compute_subnetwork" "subnetwork" {
   network     = var.network_name
   project     = var.project_id
   description = lookup(each.value, "description", null)
-  secondary_ip_range = [
-    for i in range(
-      length(
-        contains(
-        keys(var.secondary_ranges), each.value.subnet_name) == true
-        ? var.secondary_ranges[each.value.subnet_name]
-        : []
-    )) :
-    var.secondary_ranges[each.value.subnet_name][i]
-  ]
+  dynamic "secondary_ip_range" {
```

Comment:
secondary_ip_range is not a block. This will not work.

Reply:
![Uploading image.png…]()

Reply:
@jullianow thx. Will this be breaking change? What will be the effect of changing it from list to block?

Reply:
@jullianow thx. Can you plz test if some someone already created secondary IP ranges using current code, will upgrade be transparent or TF will drop and recreate secondary ranges?